metrics/ci_integrations/test/client/github_actions/test_utils/mock/github_actions_mock_server.dart

```diff
@@ -0,0 +1,237 @@
+    List<WorkflowRunArtifact> artifacts = _generateArtifacts();
+
+    artifacts = _paginate(
+      artifacts,
+      _extractPerPage(request),
+      _extractPage(request),
+    );
+
+    final _response = {
+      'total_count': artifacts.length,
+      'artifacts': artifacts.map((artifact) => artifact.toJson()).toList(),
+    };
+
+    request.response.write(jsonEncode(_response));
+
+    await request.response.flush();
+    await request.response.close();
+  }
+
+  /// Responses with the number of billable minutes and total run time
+  /// for a specific workflow run.
+  Future<void> _workflowUsageResponse(HttpRequest request) async {
+    const workflowRunDuration = WorkflowRunDuration(
+      duration: Duration(milliseconds: 500000),
+    );
+
+    request.response.write(jsonEncode(workflowRunDuration.toJson()));
+
+    await request.response.flush();
+    await request.response.close();
+  }
+
+  /// Redirects to the URL to download an archive for a repository.
+  Future<void> _downloadArtifactResponse(HttpRequest request) async {
+    final uri = Uri.parse(url);
+
+    await request.response.redirect(
+      Uri(host: uri.host, port: uri.port, path: _downloadPath),
+      status: HttpStatus.found,
+    );
+
+    await request.response.close();
+  }
+
+  /// Returns a json, containing a [Uint8List] to emulate download.
+  Future<void> _downloadResponse(HttpRequest request) async {
+    request.response.write(jsonEncode(<Uint8List>[]));
+
+    await request.response.flush();
+    await request.response.close();
+  }
+
+  /// Adds a [HttpStatus.notFound] status code to the [HttpRequest.response]
+  /// and closes it.
+  Future<void> _notFoundResponse(HttpRequest request) async {
+    request.response.statusCode = HttpStatus.notFound;
+
+    await request.response.close();
+  }
+
+  /// Chunk the given [items], limiting to the [limit],
+  /// starting from the [offset] index.
+  List<T> _paginate<T>(List<T> items, [int limit = 100, int offset = 0]) {
+    if (limit != null && offset != null) {
+      final from = (offset - 1) * limit;
+
+      return items.skip(from).take(limit).toList();
+    } else if (limit != null) {
+      return items.take(limit).toList();
+    }
+
+    return items;
+  }
+
+  /// Generates a list of [WorkflowRun], filtered by the given [RunStatus].
+  List<WorkflowRun> _generateWorkflowRuns(RunStatus status) {
+    final random = Random();
+    const statuses = RunStatus.values;
+    const conclusions = RunConclusion.values;
+
+    final runs = List.generate(100, (index) {
+      final id = index + 1;
+
+      return WorkflowRun(
+        id: id,
+        number: random.nextInt(100),
+        url: 'url',
+        status: statuses[random.nextInt(statuses.length)],
+        conclusion: conclusions[random.nextInt(conclusions.length)],
+        createdAt: DateTime.now().toUtc(),
+      );
+    });
+
+    if (status == null) {
+      return runs;
+    }
+
+    return runs.where((run) => run.status == status).toList();
```

Comment:
Maybe we can simplify it? 
```suggestion
        status: status ?? statuses[random.nextInt(statuses.length)],
        conclusion: conclusions[random.nextInt(conclusions.length)],
        createdAt: DateTime.now().toUtc(),
      );
    });

    return runs;
```